### ISSUE TITLE
Full library not indexed?

### ISSUE THREAD

**Issue by milan187** [gilesknap/gphotos-sync#262]
Hello I have a large library of 150,000 photos.

I just setup gphotos-sync on Synology NAS as a container.

![image](https://user-images.githubusercontent.com/25143355/94620323-6cd48780-027c-11eb-86a7-38866ad320cb.png)

Tried to search but did not find anything.
Bug or am I missing something?

**Comment by gilesknap** [comment]
Sorry for late reply. We have seen bugs in Google API that cause this kind of behaviour.

Are you syncing albums? Some libraries don't index all photos through raw index but then can find them in albums (this is a Google bug). Are you using library sharing? This can cause confusing results. Do you archive photos? If so then you need to tell gphotos-sync to sync archived items.

**Comment by milan187** [comment]
No worries. Thanks for reply. I'm pretty new to using this. Thank you for working on it. 

What I've noticed it that in the albums folder I see a bunch of albums but all the folders are empty. I don't necessarly use the albums, most of them are auto created. I do use a shared library with my wife. So I'm not sure if that is causing the issue. Do not use archive but it may have some photos in there like screenshot I don't care about. 

My only command right now is /storage 

How else can I approach this in terms of command to see if it will sync everything. Again I don't care about albums/archive but would like photos that are shared of kids from my wife to sync.

**Comment by gilesknap** [comment]
I'm not sure about the behaviour with shared libraries. I have looked at
this before but forget how the API tests them.

Can you set up a gphotos-sync using your wife's credentials and see if that
syncs the remaining files?

You can have two accounts co exist as long as you use two separate target
folders.

**Comment by wuhuwuhuhu** [comment]
I also use gphotos-sync on Synology NAS as a container. And I have the same problem. I think it's something wrong in the docker.

**Comment by milan187** [comment]
> I also use gphotos-sync on Synology NAS as a container. And I have the same problem. I think it's something wrong in the docker.

How are you comparing the numbers? I found out Google dashboard is wrong.

**Comment by wuhuwuhuhu** [comment]
The number is the same as the dashboard. Even when I use Python to run gphotos-sync. But when I use Google takeout, it told me I have about 800G photos in GooglePhoto, rather than 150G downloaded by gphotos-sync. I think this should be blamed on APIs. I am not sure the real number, I will use takeout to get all my photos.

**Comment by gilesknap** [comment]
@wuhuwuhuhu discrepancy in disk space is easily explained I'm afraid all the videos downloaded are trans-coded to a higher compression and much smaller file. This is loss of data. We have a long standing bug in with Google on this see #119

**Comment by milan187** [comment]
I noticed something else. As mentioned I have a shared library with wife.
I have an S20, she has Pixel 3.

All her videos get downloaded to my sync but not mine. 
Weird, any idea?

Otherwise I think I have everything.

**Comment by gilesknap** [comment]
Hi sorry for the long delay. If you are still having the missing video issue and would like to try debugging it then let me know. We would need to set up some tracing and take a look through your logs. I'm afraid that this sounds similar to other issues that have turned out to be caused by these Google bugs from #119 which they do not seem to be very receptive to fixing:
https://issuetracker.google.com/issues/151032572
https://issuetracker.google.com/issues/129250413

**Comment by Alives** [comment]
I just started using gphotos-sync and I think I ran into this issue.  While doing the original download I was watching the archive size and file download count, but ran out of quota.  After waiting 24hrs I ran it again, it continued to download, but then at the end it seems to have removed all shared content that was not originally mine.  Regardless of flags I throw at it or number of attempts, I cannot get it to sync any shared content.

I can verify that it's missing shared content by looking at a shared album from the photos.google.com website and finding a photo from someone else, looking at the filename and looking for that filename in my archive.

How can I help debug this?
